Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -591,14 +591,16 @@ private void RunMatchExpressionForSingleLineRegexes(AnalyzeContext context, Matc
             {
                 matchExpression.Properties ??= new Dictionary<string, string>();
 
+                var properties = new Dictionary<string, string>(matchExpression.Properties);
+
                 string filePath = context.TargetUri.GetFilePath();
-                matchExpression.Properties["scanTargetFullPath"] = filePath;
```

Comment:
>matchExpression. [](http://example.com/codeflow?start=16&length=16)

when analyzing in multi-thread, there were times where two different threads were changing this. When this happen, the AddProperties that is called inside `Validate` method generates an exception.

The base properties should be shared if exists.

Reply:
So, the general scan architecture design pattern for this would be to create a 'context' object per-thread. 

You can see how this would be a more efficient change, i.e., rather than copying a dictionary on every call to validate (which occurs frequently), instead, we'd copy the match expression properties a single time on thread creation, and simply update the 'scanTargetFullPath' property as we did before (because it is a thread-safe operation).